### PR TITLE
Add ConstraintGroup

### LINKED ISSUE #26: Replace Existing Constraint
Hey there,

Is it possible to replace an existing constraint? I need to do this for animation.

```swift
layout(view) { (view) in
  //...
  view.left == view.superview!.left // needs to be removed so I can change the position of the view
}
```

### PR DESCRIPTION
**Pull request by robb**
This adds `ConstraintGroup` to facilitate replacing multiple constraints at once without any behind the scenes magic.

It can currently be used like so:

```swift
constrain(view) { view in
    view.width  == 100
    view.height == 100
}

let group = ConstraintGroup()

// Attach `view` to the top left corner of its superview
layout(view, replace: group) { view in
    view.top  == view.superview!.top
    view.left == view.superview!.left
}
```

Then, later the constraints in `group` can be replaced like so:

```swift
// Move the view to the bottom right corner of its superview
constrain(view, replace: group) { view in
    view.bottom == view.superview!.bottom
    view.right  == view.superview!.right
}

UIView.animateWithDuration(0.5, animations: view.layoutIfNeeded)
```
---

`mutating` [functions can't really be used](https://github.com/robb/Cartography/commit/351b0136b912fd536025c78f56faaddcb6322985), since `ConstraintGroup` doesn't have proper value semantics, consider:

```swift
let fixed = constrain(view) {
    // bla
} 

var copy = fixed

layout(view, replace: &copy) {
    // bla bla
}

var copy2 = fixed

layout(view, replace: &copy2) {
    // bla bla
}
```

That's only going to lead to nasty surprises for people.

Closes #55 
Fixes #26 #60

### CODE REVIEW

**Comment by robb** [comment]
@devxoul what do you think about this?

**Comment by devxoul** [comment]
Nice approach! It'll be great if `ConstraintGroup` supports `NSLayoutConstraint`'s `active` property, introduced at iOS 8.
> The receiver may be activated or deactivated by manipulating this property.  Only active constraints affect the calculated layout.  Attempting to activate a constraint whose items have no common ancestor will cause an exception to be thrown. Defaults to NO for newly created constraints.

File: README.md [hunk 1]
```diff
@@ -30,6 +30,60 @@ layout(view1, view2) { view1, view2 in
...
+// Attach `view` to the top left corner of its superview
+constrain(view, replace: group) { view in
+    view.top  == view.superview!.top
+    view.left == view.superview!.left
+}
+
+/* Later */
+
+// Move the view to the bottom right corner of its superview
+group.constrain(view, replace: group) { view in
```
Comment (felixjendrusch) [inline README.md]: Did you mean `constrain(view, replace: group) { view in` (not calling `constrain` on `group`)?

Comment (robb) [inline README.md]: Yes

**Comment by robb** [comment]
These tests fail since they run on `10.9` where `NSLayoutConstraint.active` is not available.

**Pull request merged**